pkg/bitnet/internal/model/loader.go

```diff
@@ -0,0 +1,165 @@
+package model
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	ErrModelNotFound = errors.New("model file not found")
+	ErrInvalidGGUF   = errors.New("invalid GGUF magic number")
+	ErrModelNotSet   = errors.New("model path not set")
+	ErrReaderNil     = errors.New("reader is nil")
+)
+
+// GGUFHeader represents the header of a GGUF format file
+type GGUFHeader struct {
+	Magic       uint32
+	Version     uint32
+	TensorCount uint64
+	KVCount     uint64
+}
+
+// ModelLoader handles loading and managing the BitNet model file in GGUF format.
+type ModelLoader struct {
+	modelPath  string
+	bufferSize int
+	chunkPool  sync.Pool
+	header     *GGUFHeader
+}
+
+// NewModelLoader creates a new ModelLoader instance.
+func NewModelLoader() (*ModelLoader, error) {
```

Comment:
I think we should take the filesystem and model path always as an argument. This way we can support both files on the filesystem and embedded filesystems. We should always write code so that components take all global state as their arguments. It makes it easier to write tests too.